tests/unit/s2n_stuffer_network_order_test.c

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <s2n.h>
+
+#include "s2n_test.h"
+#include "stuffer/s2n_stuffer.h"
+#include "utils/s2n_mem.h"
+
+#include "stuffer/s2n_stuffer_network_order.c"
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    struct s2n_stuffer stuffer;
+
+    /* s2n_stuffer_write_network_order */
+    {
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&stuffer, 0));
+
+        /* Null checks */
+        EXPECT_FAILURE(s2n_stuffer_write_network_order(NULL, 0, 1));
+
+        /* No-op for zero length */
+        EXPECT_SUCCESS(s2n_stuffer_write_network_order(&stuffer, 0x00, 0));
+        EXPECT_EQUAL(s2n_stuffer_data_available(&stuffer), 0);
+
+        uint8_t byte_length;
+
+        /* uint8_t */
+        {
+            byte_length = sizeof(uint8_t);
+            uint8_t actual_value;
+
+            for (int i = 0; i <= UINT8_MAX; i++) {
+                EXPECT_SUCCESS(s2n_stuffer_write_network_order(&stuffer, i, byte_length));
+                EXPECT_SUCCESS(s2n_stuffer_read_uint8(&stuffer, &actual_value));
+                EXPECT_EQUAL(i, actual_value);
+            }
+        }
+
+        /* uint16_t */
+        {
+            byte_length = sizeof(uint16_t);
+            uint16_t actual_value;
+
+            for (int i = 0; i < UINT16_MAX; i++) {
```

Comment:
how long does this take?

Reply:
I didn't time it, but from a human perspective the test completes almost instantly.

Reply:
Good enough :)